pkg/cmd/cli/wait.go

```diff
@@ -0,0 +1,102 @@
+package cli
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/solo-io/solo-kit/pkg/api/v1/clients"
+	"github.com/solo-io/squash/pkg/api/v1"
+	"github.com/solo-io/squash/pkg/models"
+	"github.com/solo-io/squash/pkg/options"
+	"github.com/solo-io/squash/pkg/utils"
+	"github.com/spf13/cobra"
+)
+
+// WaitCmd
+// TODO - make this more useful, read until you find one in the Attached state
+func WaitCmd(namespace string, name string, timeout float64) (v1.DebugAttachment, error) {
+	ctx := context.Background()
+	daClient, err := utils.GetDebugAttachmentClient(ctx)
+	if err != nil {
+		return v1.DebugAttachment{}, err
+	}
+	reOptions := clients.ReadOpts{
+		Ctx: ctx,
+	}
+	da, err := (*daClient).Read(namespace, name, reOptions)
+	if err != nil {
+		// TODO(mitchdraft) implement a periodic check instead of waiting the full timeout duration
+		time.Sleep(time.Duration(int32(timeout)) * time.Second)
+		da, err = (*daClient).Read(options.SquashClientNamespace, name, reOptions)
+		if err != nil {
+			return v1.DebugAttachment{}, err
+		}
+	}
+	return *da, nil
+
+}
+
+func WaitAttCmd(o *Options) *cobra.Command {
+	// attachmentWaitTimeout := 1.0
+	waitForAttachmentCmd := &cobra.Command{
+		Use:   "wait dbgattachmentid",
+		Short: "wait for a debug config to have a debug server url appear",
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			if len(args) != 1 {
+				return fmt.Errorf("Invalid number of arguments: %v", args)
+			}
+
+			id := args[0]
+
+			reOptions := clients.ReadOpts{
+				Ctx: o.ctx,
+			}
+			timedOut := false
+			da, err := (*o.daClient).Read(options.SquashClientNamespace, id, reOptions)
+			if err != nil || da.State != v1.DebugAttachment_Attached {
```

Comment:
i would also check for error states.
if the states are in order, you can use >=

Reply:
I'm going to refactor the whole wait function when I move to the better status API